src/neuroglancer/annotation/annotation_details_tab.ts

```diff
@@ -0,0 +1,664 @@
+import {Annotation, AnnotationReference, AnnotationSource, AnnotationType, AxisAlignedBoundingBox, Collection, Ellipsoid, getAnnotationTypeHandler, Line, LineStrip, LocalAnnotationSource, Point, Spoke} from 'neuroglancer/annotation';
+import {AnnotationLayerState} from 'neuroglancer/annotation/frontend';
+import {VoxelSize} from 'neuroglancer/navigation_state';
+import {TrackableBoolean} from 'neuroglancer/trackable_boolean';
+import {WatchableValue} from 'neuroglancer/trackable_value';
+import {AnnotationSegmentListWidget, getPositionSummary, SelectedAnnotationState} from 'neuroglancer/ui/annotations';
+import {Owned} from 'neuroglancer/util/disposable';
+import {removeChildren} from 'neuroglancer/util/dom';
+import {transformVectorByMat4, vec3} from 'neuroglancer/util/geom';
+import {formatBoundingBoxVolume, formatIntegerBounds, formatLength} from 'neuroglancer/util/spatial_units';
+import {Uint64} from 'neuroglancer/util/uint64';
+import {makeCloseButton} from 'neuroglancer/widget/close_button';
+import {Tab} from 'neuroglancer/widget/tab_view';
+import {makeTextIconButton} from 'neuroglancer/widget/text_icon_button';
+import {getPreserveSourceAnnotations} from 'neuroglancer/preferences/user_preferences';
+import {StatusMessage} from 'neuroglancer/status';
+import {createPointAnnotation} from 'neuroglancer/annotation/point';
+
+const tempVec3 = vec3.create();
+type CollectionLike = AnnotationType.COLLECTION|AnnotationType.SPOKE|AnnotationType.LINE_STRIP;
+export class AnnotationDetailsTab extends Tab {
+  private valid = false;
+  private mouseEntered = false;
+  private hoverState: WatchableValue<{id: string, partIndex?: number}|undefined>|undefined;
+  private segmentListWidget: AnnotationSegmentListWidget|undefined;
+  constructor(
+      public state: Owned<SelectedAnnotationState>, public voxelSize: VoxelSize,
+      public setSpatialCoordinates: (point: vec3) => void) {
+    super();
+    this.element.classList.add('neuroglancer-annotation-details');
+    this.registerDisposer(state);
+    this.registerDisposer(voxelSize);
+    this.registerDisposer(this.state.changed.add(() => {
+      this.valid = false;
+      this.updateView();
+    }));
+    this.registerDisposer(this.visibility.changed.add(() => this.updateView()));
+    this.state.changed.add(() => {
+      this.valid = false;
+      this.updateView();
+    });
+    this.element.addEventListener('mouseenter', () => {
+      this.mouseEntered = true;
+      if (this.hoverState !== undefined) {
+        this.hoverState.value = this.state.value;
+      }
+    });
+    this.element.addEventListener('mouseleave', () => {
+      this.mouseEntered = false;
+      if (this.hoverState !== undefined) {
+        this.hoverState.value = undefined;
+      }
+    });
+    this.updateView();
+  }
+
+  private getAnnotationStateInfo() {
+    const value = this.state.value!;
+    const info = <any>{};
```

Comment:
Instead of returning any, create an interface that is local to this file and make info that type